ibeam/src/gateway_client.py

```diff
@@ -244,10 +250,12 @@ def _maintenance(self):
             _LOGGER.warning('Shutting IBeam down due to critical error.')
             self._scheduler.remove_all_jobs()
             self._scheduler.shutdown(False)
+            self._health_server.shutdown()
         elif success:
             _LOGGER.info('Gateway running and authenticated')
 
     def kill(self) -> bool:
+        self._health_server.shutdown()
```

Comment:
Since this is a blocking call, I'd suggest moving it to the end of this method

Reply:
Method has been moved